apps/token-manager/app/src/screens/Holders.js

```diff
@@ -1,166 +1,163 @@
-                        />
-                        {!groupMode && (
-                          <TableHeader
-                            title="Balance"
-                            align={compactTable ? 'left' : 'right'}
-                            colSpan={compactTable ? '2' : '1'}
-                          />
-                        )}
-                        {!groupMode && !compactTable && <TableHeader />}
-                      </TableRow>
-                    }
-                    noSideBorders={compactTable}
-                  >
-                    {holders.map(({ address, balance }) => (
-                      <HolderRow
-                        key={address}
-                        address={address}
-                        balance={balance}
-                        groupMode={groupMode}
-                        isCurrentUser={Boolean(
-                          userAccount && userAccount === address
-                        )}
-                        maxAccountTokens={maxAccountTokens}
-                        tokenDecimalsBase={tokenDecimalsBase}
-                        onAssignTokens={onAssignTokens}
-                        onRemoveTokens={onRemoveTokens}
-                        compact={compactTable}
-                      />
-                    ))}
-                  </ResponsiveTable>
-                )}
-              </Main>
-              {(!tabbedNavigation || selectedTab === 1) && (
-                <SideBar
-                  holders={holders}
-                  tokenAddress={tokenAddress}
-                  tokenDecimalsBase={tokenDecimalsBase}
-                  tokenName={tokenName}
-                  tokenSupply={tokenSupply}
-                  tokenSymbol={tokenSymbol}
-                  tokenTransfersEnabled={tokenTransfersEnabled}
-                  userAccount={userAccount}
-                />
-              )}
-            </TwoPanels>
-          )
-        }}
-      </Viewport>
-    )
-  }
+            return values
+          }}
+          renderEntryActions={([address, balance]) => (
+            <EntryActions
+              address={address}
+              balance={balance}
+              onAssignTokens={onAssignTokens}
+              onRemoveTokens={onRemoveTokens}
+              singleToken={balance.eq(tokenDecimalsBase)}
+              canAssign={balance.lt(maxAccountTokens)}
+            />
+          )}
+        />
+      }
+      secondary={
+        <InfoBoxes
+          holders={holders}
+          tokenAddress={tokenAddress}
+          tokenDecimalsBase={tokenDecimalsBase}
+          tokenName={tokenName}
+          tokenSupply={tokenSupply}
+          tokenSymbol={tokenSymbol}
+          tokenTransfersEnabled={tokenTransfersEnabled}
+          userAccount={userAccount}
+        />
+      }
+    />
+  )
+}
 
-  handleSelectTab = index => {
-    this.setState({ selectedTab: index })
-  }
+Holders.propTypes = {
+  holders: PropTypes.array,
```

Comment:
Hehe, maybe this is a good time to just list all of the props (and just declare them all to be required)?

Reply:
I added them, but I didn’t make them all required because many stay `undefined` until the data starts to be retrieved from the script.